contracts/marketplace/Marketplace.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.6.0;
+
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "../DebtToken.sol";
+
+contract Marketplace {
+  using SafeMath for uint256;
+  using SafeERC20 for DebtToken;
+
+  //Structure for sell position
+  struct sellPosition{
+    uint256 amount; //tokens that can be selled
+    uint256 sellPrice; //sell price per token in wei
+    uint256 minAmountToBuy; //minimum amount of tokens for someone to buy
+    bool    exists;
+  }
+
+  //Map of selling positions, quantity -> sellPrice(wei)
+  mapping(address => mapping( string => sellPosition)) sellPositions;
```

Comment:
Probably shouldn't use strings as loan identifiers.

Reply:
The string references the token name, this allows users to open sellPositions to more than 1 token.

Reply:
Wouldn't it be preferable to simply use the address of the ERC-20 contract managing the debt's tokens? Guaranteed uniqueness without necessity for additional identifier generation logic.